app/services/kids_user_service.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class KidsUserService
+  def initialize(share)
+    @key = share.parental_key
+    @account = share.account
+    @name = @account.name
+    @email = @account.email.blank? ? "#{@key}@budgetingkid.com" : @account.email
+  end
+
+  def perform
+    user = User.includes(:account).find_by(parental_key: @key)
+
+    return user if user
+
+    create_user
+  end
+
+  private
+
+  def create_user
+    ActiveRecord::Base.transaction do
+      user = User.create(email: @email, password: @key, parental_key: @key)
```

Comment:
Ideally, we should not destroy anything. Data is the most valuable asset in any project.

Reply:
I've caught your drift.

Reply:
It’s not a standard solution, but it makes sense.
At the same time, if a parent wants to recreate/update/recovery access and the child's email has been available since the first share, it won't be possible.